### ISSUE TITLE
There are no osqueryd messages in syslog, but logger_plugin was set correctly

### ISSUE THREAD

**Issue by macEar** [osquery/osquery#6466]
### What operating system and version are you using?

```
SELECT * FROM os_version
```

| build | codename | major | minor | name         | patch | platform | platform_like | version                                                         |
|-------|----------|-------|-------|--------------|-------|----------|---------------|-----------------------------------------------------------------|
|       |          | 7     | 8     | CentOS Linux | 2003  | rhel     | rhel          | StartFragmentCentOS   Linux release 7.8.2003 (Core) EndFragment | 

### What version of osquery are you using?

```
SELECT * FROM osquery_info

```

| build_distro | build_platform | config_hash                              | config_valid | extensions | instance_id                          | pid   | platform_mask | start_time | uuid                                 | version | watcher |
|--------------|----------------|------------------------------------------|--------------|------------|--------------------------------------|-------|---------------|------------|--------------------------------------|---------|---------|
| centos7      | 1              | c44aec2e0e45ef75e382aaf62fccfb03fd9d1af6 | 1            | active     | aa4a4117-bf09-4406-b632-da249c1ff565 | 10994 | 9             | 1590466418 | 83932242-b2b3-fb2d-49f1-3046eef4dd3a | 4.2.0   | 10991   |

### osquery configuration

```
SELECT * FROM osquery_flags
```

| description                                                                                                             | name                                    | shell_only | type   | value                                   |
|-------------------------------------------------------------------------------------------------------------------------|-----------------------------------------|------------|--------|-----------------------------------------|
| Seconds to wait for a graceful   shutdown                                                                               | alarm_timeout                           | 0          | uint64 | 4                                       |
| Allow unsafe executable   permissions                                                                                   | allow_unsafe                            | 0          | bool   | false                                   |
| Allow the audit publisher to   change auditing configuration                                                            | audit_allow_config                      | 0          | bool   | true                                    |
| Allow the audit publisher to   install file event monitoring rules                                                      | audit_allow_fim_events                  | 0          | bool   | false                                   |
| Allow the audit publisher to   install process event monitoring rules to capture fork/vfork/clone system   calls        | audit_allow_fork_process_events         | 0          | bool   | false                                   |
| Allow the audit publisher to   install process event monitoring rules                                                   | audit_allow_process_events              | 0          | bool   | true                                    |
| Allow the audit publisher to   process audit events                                                                     | audit_allow_selinux_events              | 0          | bool   | false                                   |
| Allow the audit publisher to   install socket-related rules                                                             | audit_allow_sockets                     | 0          | bool   | false                                   |
| Allow socket events to collect   domain sockets                                                                         | audit_allow_unix                        | 0          | bool   | false                                   |
| Allow the audit publisher to   install user events-related rules                                                        | audit_allow_user_events                 | 0          | bool   | false                                   |
| The audit backlog limit                                                                                                 | audit_backlog_limit                     | 0          | int32  | 4096                                    |
| The audit backlog wait time                                                                                             | audit_backlog_wait_time                 | 0          | int32  | 0                                       |
| Debug Linux audit messages                                                                                              | audit_debug                             | 0          | bool   | false                                   |
| Show debug messages for the   FIM table                                                                                 | audit_fim_debug                         | 0          | bool   | false                                   |
| Also show file accesses (not   just writes)                                                                             | audit_fim_show_accesses                 | 0          | bool   | false                                   |
| Configure the audit subsystem   from scratch                                                                            | audit_force_reconfigure                 | 0          | bool   | false                                   |
| Always uninstall all rules,   regardless of whether they were already installed or not                                  | audit_force_unconfigure                 | 0          | bool   | false                                   |
| Attempt to retain control of   audit                                                                                    | audit_persist                           | 0          | bool   | true                                    |
| Allow the audit publisher to   show partial file events                                                                 | audit_show_partial_fim_events           | 0          | bool   | false                                   |
| Shows warnings about untracked   processes (started before osquery)                                                     | audit_show_untracked_res_warnings       | 0          | bool   | false                                   |
| Directory that contains augeas   lenses files                                                                           | augeas_lenses                           | 0          | string | /usr/share/osquery/lenses               |
| AWS access key ID                                                                                                       | aws_access_key_id                       | 0          | string |                                         |
| Enable proxying of HTTP/HTTPS   requests in AWS client config                                                           | aws_enable_proxy                        | 0          | bool   | false                                   |
| Seconds between flushing logs   to Firehose (default 10)                                                                | aws_firehose_period                     | 0          | uint64 | 10                                      |
| Name of Firehose stream for   logging                                                                                   | aws_firehose_stream                     | 0          | string |                                         |
| Seconds between flushing logs   to Kinesis (default 10)                                                                 | aws_kinesis_period                      | 0          | uint64 | 10                                      |
| Enable random kinesis   partition keys                                                                                  | aws_kinesis_random_partition_key        | 0          | bool   | false                                   |
| Name of Kinesis stream for   logging                                                                                    | aws_kinesis_stream                      | 0          | string |                                         |
| AWS profile for authentication   and region configuration                                                               | aws_profile_name                        | 0          | string |                                         |
| Proxy host for use in AWS   client config                                                                               | aws_proxy_host                          | 0          | string |                                         |
| Proxy password for use in AWS   client config                                                                           | aws_proxy_password                      | 0          | string |                                         |
| Proxy port for use in AWS   client config                                                                               | aws_proxy_port                          | 0          | uint32 | 0                                       |
| Proxy HTTP scheme for use in   AWS client config (http or https, default https)                                         | aws_proxy_scheme                        | 0          | string | https                                   |
| Proxy username for use in AWS   client config                                                                           | aws_proxy_username                      | 0          | string |                                         |
| AWS region                                                                                                              | aws_region                              | 0          | string |                                         |
| AWS secret access key                                                                                                   | aws_secret_access_key                   | 0          | string |                                         |
| AWS STS session token                                                                                                   | aws_session_token                       | 0          | string |                                         |
| AWS STS ARN role                                                                                                        | aws_sts_arn_role                        | 0          | string |                                         |
| AWS STS region                                                                                                          | aws_sts_region                          | 0          | string |                                         |
| AWS STS session name                                                                                                    | aws_sts_session_name                    | 0          | string | default                                 |
| AWS STS assume role credential   validity in seconds (default 3600)                                                     | aws_sts_timeout                         | 0          | uint64 | 3600                                    |
| Maximum number of logs in   buffered output plugins (0 = unlimited)                                                     | buffered_log_max                        | 0          | uint64 | 1000000                                 |
| Size of blocks used for   POSTing data back to remote endpoints                                                         | carver_block_size                       | 0          | uint32 | 8192                                    |
| Compress archives using zstd   prior to upload (default false)                                                          | carver_compression                      | 0          | bool   | false                                   |
| TLS/HTTPS endpoint that   receives carved content after session creation                                                | carver_continue_endpoint                | 0          | string |                                         |
| Disable the osquery file   carver function (default true)                                                               | carver_disable_function                 | 0          | bool   | true                                    |
| TLS/HTTPS init endpoint for   forensic carver                                                                           | carver_start_endpoint                   | 0          | string |                                         |
| Interval to wait if reading a   configuration fails                                                                     | config_accelerated_refresh              | 0          | uint64 | 300                                     |
| Check the format of an osquery   config and exit                                                                        | config_check                            | 0          | bool   | false                                   |
| Dump the contents of the   configuration, then exit                                                                     | config_dump                             | 0          | bool   | false                                   |
| Backup config and use it when   refresh fails                                                                           | config_enable_backup                    | 0          | bool   | false                                   |
| Path to JSON config file                                                                                                | config_path                             | 0          | string | /etc/myapp/folder/osquery.conf          |
| Config plugin name                                                                                                      | config_plugin                           | 0          | string | tls                                     |
| Optional interval in seconds   to re-read configuration                                                                 | config_refresh                          | 0          | uint64 | 3600                                    |
| TLS/HTTPS endpoint for config   retrieval                                                                               | config_tls_endpoint                     | 0          | string | /api/v1/osquery/config                  |
| Number of attempts to retry a   TLS config request                                                                      | config_tls_max_attempts                 | 0          | uint64 | 3                                       |
| Set output mode to 'csv'                                                                                                | csv                                     | 1          | bool   | false                                   |
| Attempt to daemonize (POSIX   only)                                                                                     | daemonize                               | 0          | bool   | false                                   |
| Dump the contents of the   backing store                                                                                | database_dump                           | 0          | bool   | false                                   |
| If using a disk-based backing   store, specify a path                                                                   | database_path                           | 0          | string | /var/osquery/osquery.db                 |
| Add decorators as top level   JSON objects                                                                              | decorations_top_level                   | 0          | bool   | false                                   |
| Disable receiving events from   the audit subsystem                                                                     | disable_audit                           | 0          | bool   | false                                   |
| Disable scheduled query   caching                                                                                       | disable_caching                         | 0          | bool   | false                                   |
| Disable the osquery file   carver (default true)                                                                        | disable_carver                          | 0          | bool   | true                                    |
| Disable the persistent RocksDB   storage                                                                                | disable_database                        | 0          | bool   | false                                   |
| Disable log result decoration                                                                                           | disable_decorators                      | 0          | bool   | false                                   |
| Disable distributed queries   (default true)                                                                            | disable_distributed                     | 0          | bool   | false                                   |
| Disable enrollment functions   on related config/logger plugins                                                         | disable_enrollment                      | 0          | bool   | false                                   |
| Disable osquery   publish/subscribe system                                                                              | disable_events                          | 0          | bool   | false                                   |
| Disable extension API                                                                                                   | disable_extensions                      | 0          | bool   | false                                   |
| Disable atime/mtime   preservation                                                                                      | disable_forensic                        | 0          | bool   | true                                    |
| Cache calculated file hashes,   re-calculate only if inode times change                                                 | disable_hash_cache                      | 0          | bool   | false                                   |
| Disable ERROR/INFO logging                                                                                              | disable_logging                         | 0          | bool   | false                                   |
| Disable physical memory reads                                                                                           | disable_memory                          | 0          | bool   | false                                   |
| Disable re-enrollment attempts   if related plugins return invalid                                                      | disable_reenrollment                    | 0          | bool   | false                                   |
| Comma-delimited list of table   names to be disabled                                                                    | disable_tables                          | 0          | string |                                         |
| Disable userland watchdog   process                                                                                     | disable_watchdog                        | 0          | bool   | true                                    |
| Seconds between polling for   new queries (default 60)                                                                  | distributed_interval                    | 0          | uint64 | 60                                      |
| Distributed plugin name                                                                                                 | distributed_plugin                      | 0          | string | tls                                     |
| Number of times to attempt a   request                                                                                  | distributed_tls_max_attempts            | 0          | uint64 | 3                                       |
| TLS/HTTPS endpoint for   distributed query retrieval                                                                    | distributed_tls_read_endpoint           | 0          | string | /api/v1/osquery/distributed/read        |
| TLS/HTTPS endpoint for   distributed query results                                                                      | distributed_tls_write_endpoint          | 0          | string | /api/v1/osquery/distributed/write       |
| Docker UNIX domain socket path                                                                                          | docker_socket                           | 0          | string | /var/run/docker.sock                    |
| Enable userland watchdog for   extensions processes                                                                     | enable_extensions_watchdog              | 0          | bool   | false                                   |
| Enable no-op foreign virtual   tables                                                                                   | enable_foreign                          | 0          | bool   | false                                   |
| Enable numeric monitoring   system                                                                                      | enable_numeric_monitoring               | 0          | bool   | false                                   |
| Enable the syslog ingestion   event publisher                                                                           | enable_syslog                           | 0          | bool   | false                                   |
| Comma-delimited list of table   names to be enabled                                                                     | enable_tables                           | 0          | string |                                         |
| On startup, send a new   enrollment request                                                                             | enroll_always                           | 0          | bool   | false                                   |
| Name of environment variable   holding enrollment-auth secret                                                           | enroll_secret_env                       | 0          | string |                                         |
| Path to an optional client   enrollment-auth secret                                                                     | enroll_secret_path                      | 0          | string | /etc/myapp/folder/enroll_file           |
| TLS/HTTPS endpoint for client   enrollment                                                                              | enroll_tls_endpoint                     | 0          | string | /api/v1/osquery/enroll                  |
| Skip pidfile and database   state checks                                                                                | ephemeral                               | 0          | bool   | false                                   |
| Timeout to expire event   subscriber results                                                                            | events_expiry                           | 0          | uint64 | 1                                       |
| Maximum number of events per   type to buffer                                                                           | events_max                              | 0          | uint64 | 10000                                   |
| Optimize subscriber select   queries (scheduler only)                                                                   | events_optimize                         | 0          | bool   | true                                    |
| Path to a single extension to   autoload                                                                                | extension                               | 1          | string |                                         |
| Optional path to a list of   autoloaded & managed extensions                                                            | extensions_autoload                     | 0          | string | /etc/myapp/folder/extensions.load       |
| Enable INDEX on all extension   table columns (default true)                                                            | extensions_default_index                | 0          | bool   | true                                    |
| Seconds delay between   connectivity checks                                                                             | extensions_interval                     | 0          | string | 3                                       |
| Comma-separated list of   required extensions                                                                           | extensions_require                      | 0          | string |                                         |
| Path to the extensions UNIX   domain socket                                                                             | extensions_socket                       | 0          | string | /var/osquery/osquery.em                 |
| Seconds to wait for autoloaded   extensions                                                                             | extensions_timeout                      | 0          | string | 3                                       |
| Force osqueryd to kill   previously-running daemons                                                                     | force                                   | 0          | bool   | true                                    |
| List of disabled hardware   event types                                                                                 | hardware_disabled_types                 | 0          | string | partition                               |
| Size of LRU file hash cache                                                                                             | hash_cache_max                          | 0          | uint32 | 500                                     |
| Number of milliseconds to   delay after hashing                                                                         | hash_delay                              | 0          | uint32 | 20                                      |
| Toggle column headers   true/false                                                                                      | header                                  | 1          | bool   | true                                    |
| Field used to identify the   host running osquery (hostname, uuid, instance, ephemeral, specified)                      | host_identifier                         | 0          | string | hostname                                |
| Install osqueryd as a service                                                                                           | install                                 | 0          | bool   | false                                   |
| Set output mode to 'json'                                                                                               | json                                    | 1          | bool   | false                                   |
| Set output mode to 'line'                                                                                               | line                                    | 1          | bool   | false                                   |
| Set output mode to 'list'                                                                                               | list                                    | 1          | bool   | false                                   |
| Log scheduled results as   events                                                                                       | logger_event_type                       | 0          | bool   | true                                    |
| The number of acknowledgments   the leader has to receive (0, 1, 'all')                                                 | logger_kafka_acks                       | 0          | string | all                                     |
| Bootstrap broker(s) as a   comma-separated list of host or host:port (default port 9092)                                | logger_kafka_brokers                    | 0          | string | localhost                               |
| Compression codec to use for   compressing message sets ('none' or 'gzip')                                              | logger_kafka_compression                | 0          | string | none                                    |
| Kafka topic to publish logs   under                                                                                     | logger_kafka_topic                      | 0          | string |                                         |
| Minimum level for status log   recording                                                                                | logger_min_status                       | 0          | int32  | 2                                       |
| Minimum level for statuses   written to stderr                                                                          | logger_min_stderr                       | 0          | int32  | 2                                       |
| Decimal mode for log files   (default '0640')                                                                           | logger_mode                             | 0          | int32  | 416                                     |
| Use numeric JSON syntax for   numeric values                                                                            | logger_numerics                         | 0          | bool   | false                                   |
| Directory path for   ERROR/WARN/INFO and results logging                                                                | logger_path                             | 0          | string | /var/log/osquery/                       |
| Logger plugin name                                                                                                      | logger_plugin                           | 0          | string | syslog                                  |
| Log scheduled snapshot results   as events                                                                              | logger_snapshot_event_type              | 0          | bool   | true                                    |
| Always send status logs   synchronously                                                                                 | logger_status_sync                      | 0          | bool   | false                                   |
| Write status logs to stderr                                                                                             | logger_stderr                           | 0          | bool   | true                                    |
| Syslog facility for status and   results logs (0-23, default 19)                                                        | logger_syslog_facility                  | 0          | int32  | 19                                      |
| Prepend @cee: tag to logged   JSON messages                                                                             | logger_syslog_prepend_cee               | 0          | bool   | false                                   |
| GZip compress TLS/HTTPS   request body                                                                                  | logger_tls_compress                     | 0          | bool   | false                                   |
| TLS/HTTPS endpoint for results   logging                                                                                | logger_tls_endpoint                     | 0          | string |                                         |
| Max number of logs to send per   period                                                                                 | logger_tls_max_lines                    | 0          | uint64 | 1024                                    |
| Max size in bytes allowed per   log line                                                                                | logger_tls_max_linesize                 | 0          | uint64 | 1048576                                 |
| Seconds between flushing logs   over TLS/HTTPS                                                                          | logger_tls_period                       | 0          | uint64 | 4                                       |
| Log messages to stderr in   addition to the logger plugin(s)                                                            | logtostderr                             | 0          | bool   | true                                    |
| Set string for NULL values,   default ''                                                                                | nullvalue                               | 0          | string |                                         |
| File to dump numeric   monitoring records one per line. The format of the line is   <PATH><TAB><VALUE><TAB><TIMESTAMP>. | numeric_monitoring_filesystem_path      | 0          | string | /var/log/osquery/numeric_monitoring.log |
| Comma separated numeric   monitoring plugins names                                                                      | numeric_monitoring_plugins              | 0          | string | filesystem                              |
| Time period in seconds for   numeric monitoring pre-aggreagation buffer.                                                | numeric_monitoring_pre_aggregation_time | 0          | uint64 | 60                                      |
| Run all queries in a pack                                                                                               | pack                                    | 1          | string |                                         |
| Delimiter for pack and query   names                                                                                    | pack_delimiter                          | 0          | string | /                                       |
| Cache expiration for a packs   discovery queries                                                                        | pack_refresh_interval                   | 0          | uint64 | 3600                                    |
| Path to the daemon pidfile   mutex                                                                                      | pidfile                                 | 0          | string | /var/run/osqueryd.pidfile               |
| Enable osquery runtime planner   output                                                                                 | planner                                 | 1          | bool   | false                                   |
| Enable profile mode when   non-0, set number of iterations                                                              | profile                                 | 1          | int32  | 0                                       |
| Sleep a number of seconds   before and after the profiling                                                              | profile_delay                           | 0          | int32  | 0                                       |
| Optional HTTP proxy hostname                                                                                            | proxy_hostname                          | 0          | string |                                         |
| Maximum file read size                                                                                                  | read_max                                | 0          | uint64 | 52428800                                |
| Allow plugin exceptions                                                                                                 | registry_exceptions                     | 0          | bool   | false                                   |
| Max background flushes                                                                                                  | rocksdb_background_flushes              | 0          | int32  | 4                                       |
| Write buffer blocks (4k)                                                                                                | rocksdb_buffer_blocks                   | 0          | uint64 | 256                                     |
| Min write buffer number to   merge                                                                                      | rocksdb_merge_number                    | 0          | int32  | 4                                       |
| Max write buffer number                                                                                                 | rocksdb_write_buffer                    | 0          | int32  | 16                                      |
| Query interval to use if none   is provided                                                                             | schedule_default_interval               | 0          | uint64 | 3600                                    |
| Epoch for scheduled queries                                                                                             | schedule_epoch                          | 0          | uint64 | 0                                       |
| Max time drift in seconds                                                                                               | schedule_max_drift                      | 0          | uint64 | 60                                      |
| Interval in seconds to reload   database arenas                                                                         | schedule_reload                         | 0          | uint64 | 300                                     |
| Reload the SQL implementation   during schedule reload                                                                  | schedule_reload_sql                     | 0          | bool   | false                                   |
| Percent to splay config times                                                                                           | schedule_splay_percent                  | 0          | uint64 | 10                                      |
| Limit the schedule, 0 for no   limit                                                                                    | schedule_timeout                        | 0          | uint64 | 0                                       |
| Set output field separator,   default '\|'                                                                              | separator                               | 1          | string | \|                                      |
| Field used to specify the   host_identifier when set to \specified\""                                                   | specified_identifier                    | 0          | string |                                         |
| Stderr log level threshold                                                                                              | stderrthreshold                         | 0          | int32  | 2                                       |
| Timeout to expire event   subscriber results                                                                            | syslog_events_expiry                    | 0          | uint64 | 2592000                                 |
| Maximum number of events per   type to buffer                                                                           | syslog_events_max                       | 0          | uint64 | 100000                                  |
| Path to the named pipe used   for forwarding rsyslog events                                                             | syslog_pipe_path                        | 0          | string | /var/osquery/syslog_pipe                |
| Maximum number of logs to   ingest per run (~200ms between runs)                                                        | syslog_rate_limit                       | 0          | uint64 | 100                                     |
| Add an optional microsecond   delay between table scans                                                                 | table_delay                             | 0          | uint64 | 0                                       |
| Optional path to a TLS   client-auth PEM certificate                                                                    | tls_client_cert                         | 0          | string |                                         |
| Optional path to a TLS   client-auth PEM private key                                                                    | tls_client_key                          | 0          | string |                                         |
| Print remote requests and   responses                                                                                   | tls_dump                                | 0          | bool   | false                                   |
| Number of attempts to retry a   TLS enroll request, it used to be the same as [config_tls_max_attempts]                 | tls_enroll_max_attempts                 | 0          | uint64 | 3                                       |
| Override the TLS enroll secret   key name                                                                               | tls_enroll_override                     | 0          | string | enroll_secret                           |
| TLS/HTTPS hostname for Config,   Logger, and Enroll plugins                                                             | tls_hostname                            | 0          | string | my-kolide-fleet-ip:8412                 |
| Use node key as TLS endpoints                                                                                           | tls_node_api                            | 0          | bool   | false                                   |
| Include TLS enroll secret in   every request                                                                            | tls_secret_always                       | 0          | bool   | false                                   |
| Optional path to a TLS server   PEM certificate(s) bundle                                                               | tls_server_certs                        | 0          | string | /etc/app/folder/kolide.cert             |
| Reuse TLS session sockets                                                                                               | tls_session_reuse                       | 0          | bool   | true                                    |
| TLS session keep alive timeout   in seconds                                                                             | tls_session_timeout                     | 0          | uint32 | 3600                                    |
| Uninstall osqueryd as a   service                                                                                       | uninstall                               | 0          | bool   | false                                   |
| Convert all UNIX times to UTC                                                                                           | utc                                     | 0          | bool   | true                                    |
| Enable verbose informational   messages                                                                                 | verbose                                 | 0          | bool   | false                                   |
| Initial delay in seconds   before watchdog starts                                                                       | watchdog_delay                          | 0          | uint64 | 480                                     |
| Performance limit level   (0=normal, 1=restrictive, -1=off)                                                             | watchdog_level                          | 0          | int32  | 0                                       |
| Max delay in seconds between   worker respawns                                                                          | watchdog_max_delay                      | 0          | uint64 | 600                                     |
| Override watchdog profile   memory limit (e.g., 300, for 300MB)                                                         | watchdog_memory_limit                   | 0          | uint64 | 550                                     |
| Override watchdog profile CPU   utilization limit                                                                       | watchdog_utilization_limit              | 0          | uint64 | 720                                     |
| Number of work dispatch   threads                                                                                       | worker_threads                          | 0          | int32  | 4                                       |
| Time in ms to sleep after scan   of each file (default 50) to reduce memory spikes                                      | yara_delay                              | 0          | uint32 | 50                                      |
| Call malloc_trim() after YARA   scans (linux)                                                                           | yara_malloc_trim                        | 0          | bool   | true                                    |

### What steps did you take to reproduce the issue?

Start osquery the first time. Sometime the second time.

### What did you expect to see?

With the help of specified logger configuration I expect to see osqueryd messages in syslog. Every 10s the select * query is performed against process_events table which should ensure that the logs are in syslog, but there is not even one of them. 
Debug and verbose modes show that actual queries alegedly are performed as in logs we can see something like "Query process_events is performed", but nonetheless there are no logs in syslog.

### What did you see instead?

Logging to syslog starts working after osqueryd was restarted 2-4 times. Even without Kolide Fleet but with stand alone osqueryd. What could be the possible reasons for such behaivior?

Recently noticed the same behaivior on centos8.

**Comment by theopolis** [comment]
Thanks for the report, on one of the inconsistent hosts, are you able to debug further?

I'd like to see what status logs are created, to look if there is an error / warning / etc about publishing to syslog. To accomplish this can you set the logger plugin to `filesystem,syslog` this will enable both of these plugins. These logs will show up in `/var/log/osquery/` by default, with `*.WARN.*` and `*.ERROR.*` and `*.INFO.*` file names. 

If you can find an instance where osquery logs to these files, but not to syslog, can you look if there are potentially relevant messages logged?

**Comment by macEar** [comment]
Sorry for not responding for so long.

/etc/osquery/osquery.conf:
```
{}
```

/etc/osquery/osquery.flags:
```
 --enroll_secret_path=/etc/osquery/enrollment_secret
 --tls_server_certs=/etc/osquery/server.pem
 --tls_hostname=kolidefleet:8412
 --enroll_tls_endpoint=/api/v1/osquery/enroll
 --config_plugin=tls
 --config_tls_endpoint=/api/v1/osquery/config
 --disable_logging=false
 --logger_plugin=syslog
 --logger_event_type=true
 --logger_snapshot_event_type=false
 --logger_min_status=2
 --logger_min_stderr=2
```

After the problematic server has started I see that osqueryd is running with the following flags:
```
[root@XXX]# ps -edf | grep osquery
root     15071     1  0 09:24 ?        00:00:00 /usr/bin/osqueryd --flagfile /etc/osquery/osquery.flags --config_path /etc/osquery/osquery.conf
```

When I run a distributed query against `osquery_flags` table from the Kolide Fleet server, I notice that the `logger_plugin` settings is equal to `filesystem`. In Kolide configuration files there are no flags corresponding to logger settings, therefore this flag (`logger_plugin`) shouldn't have been set by Kolide Fleet. osquery might have taken the default value for this setting?

I can see the following osqueryd.INFO messages:
```
Log file created at: 2020/07/07 08:28:52
Running on machine: XXX
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0707 08:28:52.465313  1516 events.cpp:863] Event publisher not enabled: auditeventpublisher: Publisher disabled via configuration
I0707 08:28:52.466136  1516 events.cpp:863] Event publisher not enabled: syslog: Publisher disabled via configuration
```

Also want to mention (maybe this will be useful), we are using templates (images) of linux operating systems, and osquery is installed for our images.

**Comment by theopolis** [comment]
> osquery might have taken the default value for this setting?

This is what I suspect as well. Let's first deep dive into trying to reproduce this with a small test case.

**Comment by macEar** [comment]
@theopolis How can I help? What should I do for it?

The problem occures only with new servers, only when started for the first time. After I restart osqueryd by systemctl everything starts working as I can see.

May logger settings delivered by Kolide Fleet conflict with settings from osquery flags file? The settings from osquery.flags and Kolide are the same but one option: --logger_snapshot_event_type. I noticed today that this flag has different values in flags file and Kolide settings for osquery. But all other logger settings are identical, including logger_plugin which is syslog.

For now I removed all logger settings from Kolide and remained these settings only in osquery.flags file as recommended. Will see if something changes.

**Comment by theopolis** [comment]
Awesome, thanks for testing.

I'll also invest time in the near future into writing a unit test (or similar) to try to capture and reproduce this scenario. If I am able to write such a test, I should be able to write a fix as well.

**Comment by macEar** [comment]
@theopolis strange
In syslog logs I can see the following kind of events:
```
I0715 11:18:15.761612  1611 scheduler.cpp:96] Executing scheduled query pack/common/process_events: SELECT auid, cmdline, ctime, cwd, egid, euid, gid, parent, path, pid, time, uid FROM process_events;
I0715 11:07:29.662971  1611 scheduler.cpp:96] Executing scheduled query pack/common/file_events: SELECT * FROM file_events;
I0715 10:40:42.420358  1611 scheduler.cpp:96] Executing scheduled query pack/management/osquery_info: SELECT * FROM osquery_info;
```

but __no results of those queries at all in syslog__.

I have restarted osqueryd by systemctl and also tried osqueryctl but when I run query against `osquery_flags` table, I see logger_plugin option is set in one case to filesystem and in another one to syslog, but there are no events in syslog.. osquery.flags is correctly set, Kolide fleet shouldn't have touched logger settings for now (I removed from the previous time).

**Comment by macEar** [comment]
@theopolis There are some important results. Ive conducted a test on one of problematic servers.
So, osquery installed, I can issue ad-hoc queries and they work. I checked osquery.flags, no problem. I issued a query against osquery_flags table, settings are correct.
Next, i checked logs by:
```
journalctl -u osqueryd -n 10
```
And I found only 3 logs there:
```
Jul 16 08:18:03 test-server systemd[1]: Starting The osquery Daemon...
Jul 16 08:18:03 test-server systemd[1]: Started The osquery Daemon.
Jul 16 08:18:03 test-server osqueryd[508]: osqueryd started [version=4.2.0]
```

I issued another query, now against osquery_schedule table, and turns out there are no queries scheduled on that server.

After that I edited already existed pack (to update it). In 10 minutes logs are correctly transferred. Therefore it looks like a bug of Kolide Fleet...
